File: fix-duplicate-imports.py
#!/usr/bin/env python3

import re
import sys

def fix_duplicate_imports(file_path):
    """Fix duplicate imports in a file"""
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    lines = content.split('\n')
    seen_imports = set()
    cleaned_lines = []
    
    for line in lines:
        # Check if this is an import line
        import_match = re.match(r'^import\s+(\w+)\s+from', line.strip())
        if import_match:
            import_name = import_match.group(1)
            if import_name in seen_imports:
                print(f"Removing duplicate import: {import_name}")
                continue
            seen_imports.add(import_name)
        
        cleaned_lines.append(line)
    
    with open(file_path, 'w', encoding='utf-8') as f:
        f.write('\n'.join(cleaned_lines))
    
    print(f"Fixed duplicate imports in {file_path}")

if __name__ == "__main__":
    fix_duplicate_imports('/workspace/app/page.tsx')

File: fix_duplicate_imports.js
#!/usr/bin/env node

import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

function fixDuplicateImports(filePath) {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    let modified = false;
    
    // Fix duplicate React imports
    const lines = content.split('\n');
    const newLines = [];
    let hasReactImport = false;
    let reactImportLine = '';
    
    for (let i = 0; i < lines.length; i++) {
      const line = lines[i];
      
      if (line.includes("import React") && line.includes("from 'react'")) {
        if (!hasReactImport) {
          hasReactImport = true;
          reactImportLine = line;
          newLines.push(line);
        } else {
          // Skip duplicate React import
          modified = true;
        }
      } else {
        newLines.push(line);
      }
    }
    
    if (modified) {
      content = newLines.join('\n');
      fs.writeFileSync(filePath, content, 'utf8');
      console.log(`Fixed duplicate imports in: ${filePath}`);
      return true;
    }
    
    return false;
  } catch (error) {
    console.error(`Error fixing imports in ${filePath}:`, error.message);
    return false;
  }
}

function findTsxFiles(dir) {
  const files = [];
  
  function traverse(currentDir) {
    const items = fs.readdirSync(currentDir);
    
    for (const item of items) {
      const fullPath = path.join(currentDir, item);
      const stat = fs.statSync(fullPath);
      
      if (stat.isDirectory() && !item.startsWith('.') && item !== 'node_modules') {
        traverse(fullPath);
      } else if (stat.isFile() && item.endsWith('.tsx')) {
        files.push(fullPath);
      }
    }
  }
  
  traverse(dir);
  return files;
}

// Main execution
const workspaceDir = '/workspace/app';
const tsxFiles = findTsxFiles(workspaceDir);

console.log(`Found ${tsxFiles.length} TSX files to check`);

let fixedCount = 0;
for (const file of tsxFiles) {
  if (fixDuplicateImports(file)) {
    fixedCount++;
  }
}

console.log(`Fixed duplicate imports in ${fixedCount} files`);